doc/source/cluster/kubernetes/user-guides/kuberay-gcs-ft.md

```diff
@@ -0,0 +1,259 @@
+    ```python
+    import ray
+
+    @ray.remote(num_cpus=1)
+    class Counter:
+        def __init__(self):
+            self.value = 0
+
+        def increment(self):
+            self.value += 1
+            return self.value
+
+    ray.init(namespace="default_namespace")
+    Counter.options(name="counter_actor", lifetime="detached").remote()
+    ```
+
+* `increment_counter.py` is a Python script that increments the counter.
+    ```python
+    import ray
+
+    ray.init(namespace="default_namespace")
+    counter = ray.get_actor("counter_actor")
+    print(ray.get(counter.increment.remote()))
+    ```
+
+### Step 5: Create a detached actor
+
+```sh
+# Step 4.1: Create a detached actor with name `counter_actor`.
+export HEAD_POD=$(kubectl get pods --selector=ray.io/node-type=head -o custom-columns=POD:metadata.name --no-headers)
+kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/detached_actor.py
+
+# Step 4.2: Increment the counter.
+kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/increment_counter.py
+
+# 2023-09-07 16:01:41,925 INFO worker.py:1313 -- Using address 127.0.0.1:6379 set in the environment variable RAY_ADDRESS
+# 2023-09-07 16:01:41,926 INFO worker.py:1431 -- Connecting to existing Ray cluster at address: 10.244.0.17:6379...
+# 2023-09-07 16:01:42,000 INFO worker.py:1612 -- Connected to Ray cluster. View the dashboard at http://10.244.0.17:8265
+# 1
+```
+
+(kuberay-external-storage-namespace-example)=
+### Step 6: Check the data in Redis
+
+```sh
+# Step 6.1: Check the RayCluster's UID.
+kubectl get rayclusters.ray.io raycluster-external-redis -o=jsonpath='{.metadata.uid}'
+# [Example output]: 864b004c-6305-42e3-ac46-adfa8eb6f752
+
+# Step 6.2: Check the head Pod's environment variable `RAY_external_storage_namespace`.
+kubectl get pods $HEAD_POD -o=jsonpath='{.spec.containers[0].env}' | jq
+# [Example output]:
+# [
+#   {
+#     "name": "RAY_external_storage_namespace",
+#     "value": "864b004c-6305-42e3-ac46-adfa8eb6f752"
+#   },
+#   ...
+# ]
+
+# Step 6.3: Log into the Redis Pod.
+export REDIS_POD=$(kubectl get pods --selector=app=redis -o custom-columns=POD:metadata.name --no-headers)
+# The password `5241590000000000` is defined in the `redis-config` ConfigMap.
+kubectl exec -it $REDIS_POD -- redis-cli -a "5241590000000000"
+
+# Step 6.4: Check the keys in Redis.
+KEYS *
+# [Example output]:
+# 1) "864b004c-6305-42e3-ac46-adfa8eb6f752"
+
+# Step 6.5: Check the value of the key.
+HGETALL 864b004c-6305-42e3-ac46-adfa8eb6f752
+```
+
+In [ray-cluster.external-redis.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.external-redis.yaml), the `ray.io/external-storage-namespace` annotation isn't set for the RayCluster.
+Therefore, KubeRay automatically injects the environment variable `RAY_external_storage_namespace` to all Ray Pods managed by the RayCluster with the RayCluster's UID as the external storage namespace by default.
```

Comment:
This part might be confusing for a beginner.  What's the annotation for? Why do we need the environment variable?

Reply:
I see there's some explanation farther below, one option is to link to it.

Reply:
Updated https://github.com/ray-project/ray/pull/39450/commits/ef9ce5cf3bfbe794c54d27cfeca41ed56068e364